### PR TITLE
feat: XYNE-63168 compact grid layout for multiple file attachments

### PR DESCRIPTION
**Pull request by XyneSpaces**
Services-Requiring-Redeployment:

- dashboard

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

XYNE-63168. Attaching several Excel/Office/PDF files produced a tall, cluttered attachment block — each document rendered as a large preview card, one per row. Messages with more than one document file now render as a compact, responsive grid of `FilePill` cards, matching the Slack layout that was requested.

Screenshot 2026-09-10 at 1 11 24 PM

**`MessageBubble.tsx` (`AttachmentsBlock`)**

- Widen the pill trigger: `useFilePillForAllFiles = nonPreviewableFiles.length > 0 || fileAttachments.length > 1` — collapse to pills whenever there is more than one document file.
- Pill container is an auto-fill grid (`repeat(auto-fill, minmax(min(240px, 100%), 1fr))`) so the column count follows the container width. Fixed breakpoint columns track the viewport, not the message pane, which changes width when the thread panel opens.
- Pills get `max-w-none` so they fill their grid track instead of stopping at the component's `max-w-[320px]`, which previously left a wide gap down the middle of the row.
- File size is no longer passed on message pills, so the subtitle is just the file type.

**`FilePill.tsx`**

- File-type icons now come from `@xyne/icons` rather than `lucide-react`. Word, Excel and PowerPoint previously shared one `FileText` glyph and differed only by background colour, so a wall of Office files was hard to scan.
- Fuller type labels: "Excel Spreadsheet", "Word Document", "PowerPoint Presentation".
- Dropped the 40-character JS filename truncation. The container already had `truncate` + `title`, so names were being cut twice; CSS alone lets a name use the card's real width.

**`ChatListV2.utils.ts`**

- Height estimate updated to match: pills are ~62px laid out in a grid, not 256px preview cards stacked one per row. A single document with a thumbnail still estimates as a tall preview card. Without this the virtualiser over-counts message height, which the file's own comments note causes scrollbar jumps and flicker.

**Unchanged:** single-file rich preview, images, videos, HTML preview cards, attachment ordering (`position`), the "Download all" header, and click-to-open in the shared attachment viewer. Mobile keeps its existing `MobileAttachmentsGrid` path.

## Areas Touched

- [ ] `apps/backend` (API / worker)
- [x] `apps/dashboard`
- [ ] `apps/electron`
- [ ] `apps/xyne-claw` / `apps/xyne-claw-auth`
- [ ] `packages/shared` or another shared package
- [ ] Infra (docker, scripts, nix, CI)

---

## Zero (Sync) Changes

- [x] No Zero changes — **skip this section**

## Schema & Data Changes

- [x] No schema changes — **skip this section**

## Config, Environment & URLs

- [x] No config changes — **skip this section**

## API Contract

- [x] No API changes

---

## How did you test it?

Ran locally against the dev stack. Sent messages with 8 JSON files, and with a mixed set of 4 PDFs + an archive + 2 code files: both render as a packed grid that reflows with the pane width, and the same messages render correctly in the thread panel, which is narrower and so fits fewer columns. Confirmed the single-attachment path is unchanged.

Not yet verified: a message of several real `.xlsx` files, and scroll behaviour in a channel containing a many-file message (the height-estimate path).

### Automated

- [ ] Backend unit tests — `pnpm --filter xyne-spaces-backend run test`
- [ ] Claw tests — `pnpm --filter xyne-claw run test`
- [ ] End-to-end suite — `pnpm run test`
- [ ] Added / updated tests for this change
- [x] Not covered by tests 

### Manual

**Users**
- [x] Single user

**Run mode**
- [x] Local dev (`pnpm run dev:all`)

**Surface & data**
- [x] Web browser
- [x] Large / realistic data volume 

Dark theme only; light theme and narrow/mobile viewport not exercised.

---

## Checklist

- [x] Reviewed my own diff; scoped to one thing
- [ ] `pnpm run build:shared` passes
- [x] Backend `typecheck` + `build` pass 
- [x] Dashboard `lint:errors-only` + `typecheck` + `build` pass
- [ ] Formatting clean in the packages I touched
- [ ] New deps added with `pnpm --filter <pkg> add <dep>`; version pins in root `pnpm.overrides`
- [x] Commits follow `<type>: <TICKET-ID> <subject>`; branch is `fix/*` or `feature/*`
- [ ] Docs / guidelines updated where behaviour changed
- [x] No debug logging or commented-out code left behind

## Known gaps

- **Mobile parity** — `MobileAttachmentsGrid.tsx` has no `FilePill` branch and still renders full-size cards. Unchanged by this PR (0 lines touched) and not a regression, but the two renderers now differ. Follow-up ticket to be raised.
- **Tests** — see the Automated note above; there is no dashboard component-test harness to add unit tests to.

### CODE REVIEW

File: apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx [hunk 1]
```diff
@@ -193,7 +193,10 @@ const AttachmentsBlock: React.FC<AttachmentsBlockProps> = ({
   const nonPreviewableFiles = fileAttachments.filter(a => !hasDocumentThumbnail(a));
 
   // Check if we need to use FilePill for all files (mixed types)
-  const useFilePillForAllFiles = nonPreviewableFiles.length > 0;
+  // Collapse to compact pills whenever there is more than one document file,
+  // so multiple Office/Excel/PDF files render as a clean grid instead of a row
+  // of large preview cards. A single file keeps its rich preview.
+  const useFilePillForAllFiles = nonPreviewableFiles.length > 0 || fileAttachments.length > 1;
```
Comment (XyneSpaces) [inline apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx:198]: **[🟠 High] Rule 204 — add unit tests for new functionality and edge cases.** This condition (`fileAttachments.length > 1`) is the core behavior change of the PR — two or more document attachments now collapse into compact FilePills while a single document keeps its rich preview — but no tests were added to pin it. Please cover at least:
- 2+ previewable docs → FilePill grid (collapse)
- exactly 1 previewable doc, no non-previewable siblings → rich document preview preserved
- 1 previewable + 1 non-previewable doc → both render as pills (regression guard for the old `nonPreviewableFiles.length > 0` semantics)
- image/video/HTML siblings do **not** demote a single document to a pill

Per Rule 218, assert on the rendered attachment elements (FilePill / document preview) rather than just the surrounding message text. *Assumption: a component-test harness exists for dashboard components — if not, please link where this coverage will land instead.*

File: apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx [hunk 2]
```diff
@@ -359,7 +362,7 @@ const AttachmentsBlock: React.FC<AttachmentsBlockProps> = ({
           {/* Non-previewable files - use FilePill component */}
           {/* If useFilePillForAllFiles is true, show ALL files in FilePill format */}
           {(useFilePillForAllFiles ? fileAttachments : nonPreviewableFiles).length > 0 && (
-            <div className='flex flex-col gap-2'>
+            <div className='grid grid-cols-1 sm:grid-cols-2 gap-2'>
```
Comment (XyneSpaces) [inline apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx]: **[🟡 Medium] Rule 185 — verify the pixel/layout budget of the replacement layout.** `FilePill` carries `min-w-[200px]` (max `320px`), so two columns plus `gap-2` need ≥ **408px** of row width. `sm:grid-cols-2` keys off the **viewport** (≥640px), not the message container — inside a narrow container at a wide viewport (e.g. a thread side panel; *assumption — I haven't measured its actual width*) the grid still forces 2 tracks and the row overflows horizontally, where the old `flex flex-col` could never overflow. Consider a container-aware rule instead, e.g. `grid-cols-[repeat(auto-fill,minmax(min(200px,100%),1fr))]`, a container query, or relaxing FilePill's `min-w` when it renders inside this grid.

**Review by XyneSpaces — COMMENTED** [review]
Reviewed every hunk against the Xyne Spaces review guidelines — 1 high + 2 medium findings, posted as inline comments. A full summary with the recommended reviewer follows as a top-level comment shortly.

File: apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx (hunk 1 above)
Comment (XyneSpaces) [inline apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx:198]: **[🟡 Medium] R5 (duplicate/drift) — the collapse is desktop-only, and the mobile path has no equivalent.** `MobileAttachmentsGrid.tsx` (the mobile own-bubble path) still renders every attachment as a large `MessageAttachment` card and has no FilePill branch at all, so the same message will render compact on desktop and full-size on mobile. The PR description says mobile is intentionally out of scope — if that's a deliberate phase-1 decision, please note the intended follow-up here (or link a ticket); otherwise these two renderers will keep drifting. *Assumption: mobile-parity intent isn't documented anywhere visible in the diff.*

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1717

**Verdict:** Changes requested
**Reviewed:** 1 file, 2 hunks · **Findings:** 0 blocking, 1 high, 2 medium

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| 🟠 high | Rule 204 (tests for new functionality) | `apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx:199` | The new collapse behavior (`useFilePillForAllFiles`) ships with no tests — add coverage for: 2+ docs → pills, single doc → rich preview, mixed previewable/non-previewable → pills, and image/video/HTML siblings not demoting the single doc (regression guard). Assert on the file-attachment elements themselves per Rule 218. |
| 🟡 medium | Rule 185 (verify pixel sizes before replacing layout) | `apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx:365` | `sm:grid-cols-2` keys off the 640px **viewport** breakpoint, but two `FilePill`s need ≥408px (2 × `min-w-[200px]` + `gap-2`); inside a narrower message column the 2-col grid will overflow/squeeze the pills. Prefer a container-aware grid (e.g. `repeat(auto-fill, minmax(200px, 1fr))`) or a lower min width. |
| 🟡 medium | R5 (duplicate/drift) | `apps/dashboard/src/components/ui/MessageBubble/MessageBubble.tsx:199` | The collapse is desktop-only: `MobileAttachmentsGrid` still renders large `MessageAttachment` cards for multi-file, with no FilePill path — attachment presentation now diverges across surfaces. Please file a follow-up ticket for mobile parity (or include it here). |

Inline comments with full details are on review [#pullrequestreview-5158570230](https://github.com/juspay/xyne-spaces/pull/1717#pullrequestreview-5158570230) (posted on head commit `103122e`).

The diff logic itself is correct — single-file preview preserved, ordering untouched, edge cases (mixed previews, image/video/HTML exclusion) check out; frontend-only, no security concerns.

### Recommended reviewer
**@harshpreet931** (Harshpreet Singh) — tagged here; please also add him as a reviewer on GitHub (see note below).

| Candidate | Owns (files in this PR) | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-------------------------|-------------------|-----------------|-------------|
| Anurag Dwivedi (`anurag.dwivedi@juspay.in`) | Owns the attachment-UI ticket cluster in Spaces, named first in the feature-origin thread; no commits found on `MessageBubble.tsx` | No | ~11 (Spaces PR-review pings) | Strongest domain context, but **not resolvable to a GitHub login** → can't be requested/@mentioned here |
| **@harshpreet931** (Harshpreet Singh) | Adjacent: PRs #414/#415 touched `FileViewerModal` + `ThreadList` (file-attachment surfaces), not `MessageBubble.tsx` | No | ~9–10 (Spaces pings, mostly unanswered DMs) | **Only non-requester taggable in both systems**; active today |
| @Dnyaneshwari-04 | Strongest file ownership — PR #920 assignee on the exact `AttachmentsBlock` region | No | 0 | Excluded — she is the requester driving this PR |
| Harsh Sharma | Named in the feature-origin pool | No | 0 found | Not resolvable to a GitHub login |

**Signal notes:** `.github/CODEOWNERS` only covers ACL paths — no default reviewer applies to this file, so ranking is file-ownership + Spaces context only. Pending-review counts come from Spaces PR-review pings (GitHub lexical search doesn't honor `review-requested:`). **Assignment limitation:** no reviewer-request API tool was available this run (the same wall earlier runs hit with Anurag), so @harshpreet931 is tagged via this comment — add him as a reviewer on GitHub. Anurag is the domain owner; add him manually too if you want him on it.

**Rationale:** @harshpreet931 is the only reviewer reachable in both systems (GitHub login `harshpreet931` + Spaces `harshpreet.singh@juspay.in`), has shipped adjacent file-attachment UI work, and is active today; his queue is lighter than Anurag's (~9–10 vs ~11 pending, and mostly stale DMs rather than active review requests).

**Review by harshpreet931 — APPROVED** [review]

**Review by Mohan0544 — APPROVED** [review]
